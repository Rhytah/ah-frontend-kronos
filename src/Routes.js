import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Router } from 'react-router';
import SignUpForm from './components/auth/registration/SignUpForm';
import AuthNavigation from './components/common/navigation/AuthNavigation';
import UpdateProfileForm from './components/profile/UpdateProfileForm';
import ViewProfileForm from './components/profile/ViewProfileForm';
import history from './utils/history';
import CreateArticle from './components/articles/CreateArticle';
import EditArticle from './components/articles/EditArticle';
import LoginContainer from './components/auth/login/loginContainer';
import Articles from './components/articles/articlesComponent';
import Article from './components/articles/articleComponent';
import Footer from './components/common/Footer/Footer';
import PrivateRoute from './utils/privateRoute';
import AuthorProfile from './components/profile/authorProfile';

function Routes() {
  return (
    <Router history={history}>
      <AuthNavigation />
      <Switch>
        <Route exact path="/" component={Articles} />
        <Route path="/signup" component={SignUpForm} />
        <PrivateRoute
          exact
          path="/editProfile"
          component={UpdateProfileForm}
          isAuthenticated={sessionStorage.getItem('isLoggedIn')}
        />
        <PrivateRoute
          exact
          path="/viewProfile"
          component={ViewProfileForm}
          isAuthenticated={sessionStorage.getItem('isLoggedIn')}
        />
        <Route path="/login" component={LoginContainer} />
        <PrivateRoute
          path="/update-article"
          component={EditArticle}
          isAuthenticated={sessionStorage.getItem('isLoggedIn')}
        />
        <PrivateRoute
          path="/create-article"
          component={CreateArticle}
          isAuthenticated={sessionStorage.getItem('isLoggedIn')}
        />
        <PrivateRoute
          path="/articles"
          component={Articles}
          isAuthenticated={sessionStorage.getItem('isLoggedIn')}
        />
        <PrivateRoute
          exact
          path="/profile/:username"
          component={AuthorProfile}
          isAuthenticated={sessionStorage.getItem('isLoggedIn')}
        />
        <Route path="/:slug" component={Article} />
      </Switch>
      <Footer />
    </Router>
  );
}

export default Routes;
