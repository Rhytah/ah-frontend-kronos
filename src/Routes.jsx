/* eslint-disable import/no-named-as-default */
import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Router } from 'react-router';
import SignUpForm from './components/auth/registration/SignUpForm';
import AuthNavigation from './components/common/navigation/AuthNavigation';
import history from './utils/history';
import CreateArticle from './components/articles/CreateArticle';
import EditArticle from './components/articles/EditArticle';
import LoginContainer from './components/auth/login/loginContainer';
import Articles from './components/articles/articlesComponent';
import Article from './components/articles/articleComponent';

function Routes() {
  return (
    <Router history={history}>
      <AuthNavigation />
      <Switch>
        <Route exact path="/" component={Articles} />
        <Route path="/signup" component={SignUpForm} />
        <Route path="/login" component={LoginContainer} />
        <Route path="/update-article" component={EditArticle} />
        <Route path="/create-article" component={CreateArticle} />
        <Route path="/articles" component={Articles} />
        <Route path="/:slug" component={Article} />
      </Switch>
    </Router>
  );
}

export default Routes;
