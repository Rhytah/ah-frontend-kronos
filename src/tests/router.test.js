import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';

import Routes from '../Routes';


describe('<Routes />', () => {
  const wrapper = shallow(<Routes />);
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
