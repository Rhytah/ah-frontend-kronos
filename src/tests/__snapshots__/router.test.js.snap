// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> matches snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(Navigation) />
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Articles)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(SignUpForm)",
          "type": [Function],
        }
      }
      path="/signup"
    />
    <PrivateRoute
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(UpdateProfile)",
          "type": [Function],
        }
      }
      exact={true}
      isAuthenticated={null}
      path="/editProfile"
    />
    <PrivateRoute
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(ViewProfile)",
          "type": [Function],
        }
      }
      exact={true}
      isAuthenticated={null}
      path="/viewProfile"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(LoginContainer)",
          "type": [Function],
        }
      }
      path="/login"
    />
    <PrivateRoute
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(EditArticle)",
          "type": [Function],
        }
      }
      isAuthenticated={null}
      path="/update-article"
    />
    <PrivateRoute
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(CreateArticle)",
          "type": [Function],
        }
      }
      isAuthenticated={null}
      path="/create-article"
    />
    <PrivateRoute
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Articles)",
          "type": [Function],
        }
      }
      isAuthenticated={null}
      path="/articles"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Article)",
          "type": [Function],
        }
      }
      path="/:slug"
    />
  </Switch>
  <Footer />
</Router>
`;
